### PR TITLE
Several changes

### PR DESCRIPTION
**Pull request by haibinxie**
1. Do not run retain-ds-resource hook when ds is not present
2. add pre-upgrade to service account hook resources
3. do not add migration approval annotation when daemonset is not present
4. add labels to stc for helm managed resource
5. always populate env variable so it wont fail when env is null 
6. do not populate spec.placement when it's empty to prevent failure

### CODE REVIEW

File: community/portworx/templates/storage-cluster.yaml [hunk 1]
```diff
@@ -42,8 +42,17 @@ metadata:
     portworx.io/pvc-controller: "true"
     {{- end }}
     portworx.io/is-iks: "true"
+    {{- if lookup "apps/v1" "DaemonSet" "kube-system" "portworx" }}
```
Comment (piyush-nimbalkar) [inline community/portworx/templates/storage-cluster.yaml:45]: We can do a small optimization on the first line of this file - 
```
{{- if .Values.generateStorageClusterForMigration }}
```
We want to always deploy the storage cluster irrespective of the daemonset. We don't want to deploy it only if the customer explicitly sets `generateSTCForMigration` to false.

nit: Can we rename `generateSTCForMigration` to `generateStorageClusterForMigration`? It won't be very clear to everyone what STC means.

File: community/portworx/templates/storage-cluster.yaml [hunk 2]
```diff
@@ -134,10 +143,8 @@ spec:
           key: "password"
           name: "{{ $etcdSecret }}"
   {{- end }}
-  {{- if eq $csiCloudDrive true }}
-    - name: ENABLE_CSI_DRIVE
-      value: "{{ $csiCloudDrive }}"
-  {{- end }}
+  - name: ENABLE_CSI_DRIVE
```
Comment (piyush-nimbalkar) [inline community/portworx/templates/storage-cluster.yaml:146]: Why set the variable if the value is false or not present?

**Review by piyush-nimbalkar — APPROVED** [review]

File: community/portworx/templates/storage-cluster.yaml (hunk 2 above)
Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml:146]: it complains env cannot be "null". and it's hard to enumerate all cases to decide whether env is null or not so with this env will never be null.

File: community/portworx/templates/storage-cluster.yaml (hunk 1 above)
Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml:45]: It does not sound right considering this scenario, customer first upgrade from ds to operator with generateStorageClusterForMigration set to true, then change px version and perform an operator to operator upgrade. in this case we will add the annotation back.

Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml:45]: generateStorageClusterForMigration is only used when daemonset is present, and when daemonset is not present, this flag is ignored.

Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml:45]: BTW, the comment says StorageCluster so it should be fine, i am ok changing it to generateStorageClusterForMigration to be consistent if you changed it in generic chart.

**Pull request merged**